spec/unit/chef_fs/file_system_spec.rb

```diff
@@ -145,4 +147,32 @@
       end
     end
   end
+
+  context "copy_to" do
+    let(:chef_fs_config) {Chef::ChefFS::Config.new}
+    let(:src_root) { memory_fs("/Users/vsatish/Desktop/projects/chef", {}) }
+    let(:dest_root) { memory_fs("/Users/vsatish/Desktop/projects/chef/copy_dest_test", {}) }
+
+    it "when is folder is empty copy" do
+      puts "\n src_root = #{src_root}\n\n"
+      puts "\n chef_fs_config = #{chef_fs_config}\n\n"
+      puts "\n chef_fs_config.chef_fs = #{chef_fs_config.chef_fs}\n\n"
+      puts "\n chef_fs_config.local_fs = #{chef_fs_config.local_fs}\n\n"
+
+      top_level_paths = chef_fs_config.chef_fs.children.map { |entry| entry.path }
+      puts "\n top_level_paths = #{top_level_paths}\n\n"
+
+      # top_level_paths.each do |path|
+      #   chef_fs_copy_pattern(path, chef_fs_config, name)
+      # end
+
+      error, result = Chef::ChefFS::FileSystem.copy_to(
+        Chef::ChefFS::FilePattern.new('/client'), chef_fs_config.chef_fs, chef_fs_config.local_fs, "", {})
+
+      puts "error = #{error}\n\nresult = #{result}\n\n"
```

Comment:
We should assert on the error/result matching what we expect it to.

Reply:
Its highly difficult to call the copy_to method from the test case. The way we are calling currently will return empty result struct. should we validate the result with empty struct? Please verify if it is possible to add the test case for copy_to